packages/google_sign_in/lib/google_sign_in.dart

```diff
@@ -15,6 +15,30 @@ export 'widgets.dart';
 
 enum SignInOption { standard, games }
 
+/// Signals Google authentication errors that can be recovered with user action.
+///
+/// Is thrown only on android.
+///
+/// Dart equivalent of android `UserRecoverableAuthException`.
+class AndroidUserRecoverableAuthException implements Exception {
```

Comment:
Oh sorry. I think I misunderstood your request then. Did you want a functionality similar to this, but for iOS?

Reply:
My team doesn't need it for iOS; however, I assume the same problem exists on iOS, and I'm asking if it might be a good idea to not scope this API specifically to Android to avoid a breaking change in the future if support for iOS were added. To be clear, this pull request probably does what we need (still have to look over it more thoroughly).